src/arrayMethodJoin.js

```diff
@@ -4,8 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    let separatorCopy = separator;
+
+    if (separator === undefined) {
+      separatorCopy = '';
+    } else if (separator === null) {
+      separatorCopy = 'null';
+    }
```

Comment:
you don't need this as you have already set the default value `separator = ',`'

Reply:
Ok. I`ve deleted 11-12 rows, but without next my code doesn`t pass tests. Or is that enough?)